routing/single_vehicle_world_graph.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include "routing/cross_mwm_graph.hpp"
+#include "routing/edge_estimator.hpp"
+#include "routing/geometry.hpp"
+#include "routing/index_graph.hpp"
+#include "routing/index_graph_loader.hpp"
+#include "routing/num_mwm_id.hpp"
+#include "routing/road_graph.hpp"
+#include "routing/segment.hpp"
+#include "routing/world_graph.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include <memory>
+#include <vector>
+
+namespace routing
+{
+class SingleVehicleWorldGraph final : public WorldGraph
+{
+public:
+  SingleVehicleWorldGraph(std::unique_ptr<CrossMwmGraph> crossMwmGraph,
+                          std::unique_ptr<IndexGraphLoader> loader,
+                          std::shared_ptr<EdgeEstimator> estimator);
+
+  // |isEnding| == true iff |segment| is first or last segment of the route. Needed because first
+  // and last segments may need special processing.
+  void GetEdgeList(Segment const & segment, bool isOutgoing, bool isLeap, bool isEnding,
```

Comment:
nit: consider to add a comment:
```c++
// WorldGraph overrides:
```
to this group of overrides.

Reply:
done